### PR TITLE
Show commit status icon in commits table

### PR DESCRIPTION
**Pull request by lafriks**
This would be the first step to display status icons in commit table

### Screenshot:
attels

### CODE REVIEW

**Comment by lunny** [comment]
`TargetURL` could be a link opened on a new tab. And it seems `CommitStatusState` need one more value `CommitStatusNone` if there is no status for a commit.

**Comment by lunny** [comment]
@bkcsoft

**Comment by lafriks** [comment]
Fixed also bug in https://github.com/go-gitea/gitea/pull/1688/commits/d066d05512646f4d13d0da3aeae4219e823e5b5c that it was failing render commits table when creating PR with signed commits

**Comment by lunny** [comment]
Great enough and LGTM. Maybe add an integration test for the commit table page is better!

**Comment by lunny** [comment]
Let L-G-T-M work

**Comment by lafriks** [comment]
Added integration tests for commit table and all commit statuses

**Comment by lafriks** [comment]
In master getting latest status using API does not work except for sqlite. Fixed this here in latest commit

File: integrations/repo_commits_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+package integrations
+
+import (
+	"bytes"
+	"net/http"
+	"path"
+	"testing"
+
+	//"fmt"
```
Comment (lunny) [inline integrations/repo_commits_test.go]: unused comment

File: models/status.go [hunk 2]
```diff
@@ -145,9 +146,27 @@ func GetCommitStatuses(repo *Repository, sha string, page int) ([]*CommitStatus,
...
+		Where("repo_id = ?", repo.ID).And("sha = ?", sha).
+		Select("context, max(created_unix) as created_unix").
+		GroupBy("context").OrderBy("max(created_unix) desc").Find(&statuses)
+	if err != nil {
+		return nil, err
+	}
+	fullStatuses := make([]*CommitStatus, 0, len(statuses))
+	for _, status := range statuses {
+		cs := make([]*CommitStatus, 0, 1)
+		err = x.Where("repo_id = ?", repo.ID).And("sha = ?", sha).
```
Comment (lunny) [inline models/status.go]: Why need query it twice?

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by appleboy** [comment]
Testing with https://github.com/drone/drone/pull/2017 PR. It is working for me.

screen shot 2017-05-07 at 10 54 28 am

File: models/status.go (hunk 2 above)
Comment (lafriks) [reply models/status.go]: Because there is no way to get newest row grouped by context in one select that would work across all supported databases

Comment (lunny) [reply models/status.go]: I suggest 
```Go
var ids []int64
err := x.Limit(10, page*10).
  Where("repo_id = ?", repo.ID).And("sha = ?", sha).
  Select("id").
  GroupBy("context").OrderBy("max(created_unix) desc").Find(&ids)
if len(ids) > 0 {
x.In("id", ids).Find(&statuses)
}
```

Comment (lafriks) [reply models/status.go]: You can't select columns that are outside of group by expressions

**Comment by lafriks** [comment]
Ok, I have rewritten latest commit status selects. Only drawback that it does not selects it by latest created_at column but by largest id. But that is sacrifice that can be taken for performance reasons

**Review by lunny — APPROVED** [review]

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by appleboy** [comment]
make LGTM work

**Comment by lunny** [comment]
The next step maybe PR's commit status or PR's status?

**Comment by lafriks** [comment]
I think PR commit table is same as repository commit table, so there already should be commit status. Next step would probably be PR status. Is it just PR latest commits status?

**Comment by lunny** [comment]
Yes. I think PR status == PR latest commit status

**Comment by tonivj5** [comment]
Now https://github.com/drone/drone/pull/2017 has been merged, is the PR's status the next step?

**Pull request merged**